File: pkg/server/pokete/msg/player/player.go
package player

import (
	"github.com/lxgr-linux/pokete/bs_rpc/msg"
	"github.com/lxgr-linux/pokete/server/pokete/user"
)

const PlayerType msg.Type = "pokete.player.player"

type Player struct {
	msg.BaseMsg
	User user.User `json:"user"`
}

func (p Player) GetType() msg.Type {
	return PlayerType
}

func NewPlayer(user user.User) Player {
	return Player{msg.BaseMsg{}, user}
}

File: pokete_classes/multiplayer/msg/player/player.py
from typing import TypedDict

import bs_rpc
from pokete_classes.multiplayer.msg.poke import PokeDict
from pokete_classes.multiplayer.msg.position.update import Position

PLAYER_TYPE = "pokete.player.player"


class User(TypedDict):
    name: str
    position: Position
    client: None
    pokes: list[PokeDict]
    items: dict[str, int]

class PlayerData(TypedDict):
    user: User


class Player(bs_rpc.Body):
    def __init__(self, data: PlayerData):
        super().__init__(PLAYER_TYPE, data)
